chapters/grammar.adoc

```diff
@@ -297,9 +286,6 @@ _identifier_list_ : ::
     _COMMA_ _IDENTIFIER_ +
     _identifier_list_ _COMMA_ _IDENTIFIER_
 
-_function_prototype_ : ::
-    _function_declarator_ _RIGHT_PAREN_
-
```

Comment:
I've made a different change to remove this annoying rule in #252. We do need to keep _function_prototype_ (it's referred to in _function_definition_, above), but if we structure it properly then this step isn't needed at all.

Reply:
Ah, good catch, I'd missed that one earlier because `function_definition` was weirdly at the end of the grammar instead of with the other functions.

Reply:
I merged it into `function_defintion` to keep this Cl correct, but either way these rules are very confusing to read.